shell/browser/api/electron_api_web_contents.cc

```diff
@@ -2003,6 +2002,21 @@ void WebContents::LoadURL(const GURL& url,
 
   // Required to make beforeunload handler work.
   NotifyUserActivation();
+
+  // For transparent windows, set the background color of RenderWidgetHostView.
+  // We need to set this color after LoadURL, because the RenderViewHost is only
+  // created after loading a page; a default white color is rendered otherwise.
```

Comment:
Hm, I'm not sure `LoadURL` is quite the right place to do this. If we need to do this for every render view in the window, perhaps `RenderViewHostChanged` is the right place? That would additionally account for cross-process navigations.

I'm not totally sure that this will exercise the code path I'm imagining as I haven't pulled down this PR to test locally, but here's a fiddle that I think might tickle some bad behavior with this PR applied: https://gist.github.com/5aef6fdac4379c36df0bc7b9aa8aecf7

1. It opens a child window pointing at a data: url (when you click the button)
2. the child window is transparent
3. clicking the "x-process nav" link will result in a change of RenderWidgetHostView, and on current 12.0.0 results in a flash of white.

If the above fiddle doesn't produce a flash of white with this PR applied, then we're probably safe doing this here in LoadURL, but if it **does** flash white, we should probably move this code to RenderViewHostChanged (or some other similar method that also gets called when a process swap happens).

Reply:
also: just saw that in 13.x this was moved to RenderFrameCreated, which is probably not exactly correct as it will be called when child frames are created as well as the main frame, and we don't need to update the background color when a child frame is created. it _would_ get called for x-process navigations though, which explains why the symptom is fixed in 13.

Reply:
Oh awesome, thanks so much for making a Fiddle too! 🙂 Will give this a try and report back - if it works, I'll go ahead and move the code over to RenderViewHostChanged 🙇‍♀️

Reply:
Update: it looks as though moving the setBackgroundColor logic to either `RenderViewHostChanged` or `RenderFrameHostChanged` doesn't solve the problem. Both are triggered in the code path (logging verified this), but neither seem to set the background color for transparency correctly. After double checking 13-x-y and master, it looks as though the issue is still present in both of those branches as well.

The bug seems to happen in a when a window.open method is triggered from an existing BrowserWindow, and creates a new, transparent BrowserWindow overlaying the previous one. When trying to load an external site, like Github or CNN, there is a brief white flash present regardless of #27593's fix. (Reading previous conversations, it [may not be possible to completely get rid of that flashing](https://github.com/electron/electron/issues/4985#issuecomment-204941449) on the webContents side.)

I'm a little puzzled on why `LoadURL` specifically is correctly setting the background color, but other methods (which are also firing!) are not. `LoadURL` _after_ `RenderViewCreated`, so this may be a race condition where our RenderViewHost hasn't been created yet? Both `RenderViewHostChanged` and `RenderFrameHostChanged` do fire after `LoadURL`, but don't seem to have an affect on the existing webContents 🤔

Reply:
@VerteDinde Do you have a consistent fiddle repro for this?